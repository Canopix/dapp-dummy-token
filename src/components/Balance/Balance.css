#balance-container {
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: center;
}
#balance-container p {
  margin: 0.5rem 0;
}
.balance-card {
  padding: 1rem;
}
