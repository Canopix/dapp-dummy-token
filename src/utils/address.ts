export const shortAddress = (address:string) => {
    const ln = address.length;
    return `${address.slice(0,6)}...${address.slice(ln - 4, ln)}`
}   