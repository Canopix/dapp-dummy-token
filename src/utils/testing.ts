import * as easyPeasy from 'easy-peasy'
import { models } from '../store'

export const useStoreActionsFunction = jest.fn(() => ({
  setAddress: jest.fn(),
  setChainId: jest.fn(),
  setLoading: jest.fn(),
  setIsInjected: jest.fn(),
  setBalance: jest.fn(),
  requestAccount: jest.fn(),
  requestChainId: jest.fn(),
  requestBalance: jest.fn(),
  sendCoins: jest.fn(),
}))

export const spyOnStoreActions = () =>
  jest
    .spyOn(easyPeasy, 'useStoreActions')
    .mockImplementation(useStoreActionsFunction)

export const walletModel = {
  address: '0xc77DD46aBFf64BB1dE74e6419e77258Bdc98622a',
  chainId: '0x3',
  loading: false,
  isInjected: false,
  balance: 500,
  contractAddress: '0xCa498a756ea7f05BBa1958542C2CD8B7C99AFcB3',
}

export const mockStore = easyPeasy.createStore(models, {
  initialState: {
    wallet: { ...models.wallet, ...walletModel },
  },
})

export const storeOnDemand = (initialValues: Object) => easyPeasy.createStore(models, {
    initialState: {
      wallet: { ...models.wallet, ...walletModel, ...initialValues },
    },
  })

