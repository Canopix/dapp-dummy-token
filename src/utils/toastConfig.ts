export const SUCCESS_MSG = 'Transaction was successful!'
export const ERROR_TX = 'There was an error with the transaction. Would you please try it again?'

export const DEFAULT_DURATION = 5000;
