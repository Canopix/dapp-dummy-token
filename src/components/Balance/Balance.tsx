import React from 'react'
import { useStoreState } from '../../store/index'
import { Link } from 'react-router-dom'
import { Card, Header } from 'decentraland-ui'
import { shortAddress } from '../../utils/address'
import './Balance.css'

const Balance = () => {
  const { address, balance } = useStoreState((state) => state.wallet)

  return (
    <div id="balance-container">
      <Card centered>
        <Card.Content>
          <Header size="medium">Wallet</Header>
          <p>
            <strong>Address:</strong> {shortAddress(address)}
          </p>
          <p>
            <strong>Balance:</strong> {balance} DUMMY{' '}
            <Link to="/transfer">TRANSFER</Link>
          </p>
        </Card.Content>
      </Card>
    </div>
  )
}

export default Balance
