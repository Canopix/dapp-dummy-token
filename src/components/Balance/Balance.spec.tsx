import { mount, configure } from 'enzyme'
import { StoreProvider } from 'easy-peasy'
import Balance from './Balance'
import { Router, Link } from 'react-router-dom'
import { mockStore as store } from '../../utils/testing'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { createBrowserHistory } from 'history'

const history = createBrowserHistory()
configure({ adapter: new Adapter() })

describe('Home Component', () => {
  const subject = (
    <StoreProvider store={store}>
      <Router history={history}>
        <Balance />
      </Router>
    </StoreProvider>
  )
  it('match snapshot', () => {
    const wrapper = mount(subject)
    expect(wrapper).toMatchSnapshot()
  })
  it('should have a minified address', () => {
    const wrapper = mount(subject)
    expect(wrapper.text().includes('0xc77D...622a')).toBeTruthy()
  })
  it('should present a text of 500 DUMMY on the balance', () => {
    const wrapper = mount(subject)
    expect(wrapper.text().includes('500 DUMMY')).toBeTruthy()
  })
  it('should be a transfer link', () => {
    const wrapper = mount(subject)
    expect(wrapper.find(Link).props().to).toBe('/transfer')
  })
})
