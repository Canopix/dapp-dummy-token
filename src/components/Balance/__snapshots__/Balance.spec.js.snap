// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component match snapshot 1`] = `
<StoreProvider
  store={
    Object {
      "@@observable": [Function],
      "addModel": [Function],
      "clearMockedActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getListeners": [Function],
      "getMockedActions": [Function],
      "getState": [Function],
      "persist": Object {
        "clear": [Function],
        "flush": [Function],
        "resolveRehydration": [Function],
      },
      "reconfigure": [Function],
      "removeModel": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Balance>
      <div
        id="balance-container"
      >
        <t
          centered={true}
        >
          <div
            className="ui centered card"
            onClick={[Function]}
          >
            <Ls>
              <div
                className="content"
              >
                <_a
                  size="medium"
                >
                  <div
                    className="ui medium header"
                  >
                    Wallet
                  </div>
                </_a>
                <p>
                  <strong>
                    Address:
                  </strong>
                   
                  0xc77D...622a
                </p>
                <p>
                  <strong>
                    Balance:
                  </strong>
                   
                  500
                   DUMMY
                   
                  <Link
                    to="/transfer"
                  >
                    <LinkAnchor
                      href="/transfer"
                      navigate={[Function]}
                    >
                      <a
                        href="/transfer"
                        onClick={[Function]}
                      >
                        TRANSFER
                      </a>
                    </LinkAnchor>
                  </Link>
                </p>
              </div>
            </Ls>
          </div>
        </t>
      </div>
    </Balance>
  </Router>
</StoreProvider>
`;
